Sources/TuistKit/Services/Inspect/InspecImplicitImportsService.swift

```diff
@@ -46,41 +72,74 @@ final class InspectImplicitImportsService {
         self.targetScanner = targetScanner
     }
 
-    func run(path: String?) async throws {
+    func run(
+        path: String?,
+        inspectType: InspectType
+    ) async throws {
         let path = try self.path(path)
         let config = try await configLoader.loadConfig(path: path)
         let generator = generatorFactory.defaultGenerator(config: config, sources: [])
         let graph = try await generator.load(path: path)
-        let issues = try await lint(graphTraverser: GraphTraverser(graph: graph))
+        let issues = try await lint(graphTraverser: GraphTraverser(graph: graph), inspectType: inspectType)
         guard issues.isEmpty else {
-            throw InspectImplicitImportsServiceError.implicitImportsFound(issues)
+            switch inspectType {
+            case .redundant:
+                throw InspectRedundantImportsServiceError.redundantImportsFound(issues)
+            case .implicit:
+                throw InspectImplicitImportsServiceError.implicitImportsFound(issues)
```

Comment:
These errors suggest there are two different services for inspecting imports. That is not true – we currently have only one.